content/cucumber/step-definitions.md

```diff
@@ -253,3 +253,60 @@ Given("I have {int} {color} balls", function (int1, color) {
 });
 ```
 {{% /block %}}
+
+# How do I group my step definitions?
+
+Avoid writing similar step definitions which can lead to clutter. While documenting your steps helps, making use of **helper methods** to abstract them can do wonders.
+
+For eg, if you have the following steps :
+
+```
+    Given I go to the home page
+    Given I check the about page of the website 
+    Given I get the contact details
+```
+
+and all of these steps visit the **Home**, **About** and **Contact** pages, you might be writing _redundant steps_. While the underlying code for these steps could be different, their **behaviour** is essentially the same, ie.
+
+1. Open the Home page.
+2. Open the About page.
+3. Open the Contact page.
+
+As such, you can use abstract helper methods to reduce all these into one step :
+
+    Given I go to the {} page
+
+with the step def :
+
+```java
+@Given("I want to open the {string} page")
+public void i_want_to_open_page(String name) {
+  pageFactory.openPage(name);
+}
+```
+
+```javascript
+ //TODO
+```
+
+```ruby
+ //TODO
+```
+
+
+```kotlin
+ //TODO
+```
+
+And have your step definitions act as a gateway to the actual code (in this case, a factory method to decide which page to open).
```

Comment:
"And have your step definitions act as a gateway to the actual code (in this case, a factory method to decide which page to open)." -> Suggestion: "Your step definitions are the glue to the actual code (in this example, a factory method to decide which page to open)."

You could also have a look at the example in the [FAQ](https://docs.cucumber.io/community/faq/#how-to-use-helper-methods); it goes nicely with yours. You might also want to have step definitions that hide a bunch of implementation details (like how to log in) and/or combine a few of these reusable methods (login, go to page, etc).

Reply:
I was going to add a line to emphasize "hiding implementation details by having one step call several helper methods" as well but the FAQ already covers it pretty well so I decided against it.

Reply:
It would be my preference to clearly explain in the docs themselves, and provide a summary of key points in the FAQ, pointing to the relevant section(s) of the docs from there. So a little duplication wouldn't hurt imho. But let's see what we end up with. (Incrementally "refactor" the docs :))

Reply:
Alright. I ll put it in then. Let me know what you think.